### PR TITLE
Refactor reducers that keep the dimension

### PR DESCRIPTION
**Pull request by phofl**
This makes the abstraction a bit nicer, since these reducers behave different than all the others

### CODE REVIEW

**Comment by mrocklin** [comment]
No immediate thoughts.  @rjzamora any thoughts?

File: dask_expr/reductions.py [hunk 1]
```diff
@@ -408,7 +368,49 @@ def aggregate_kwargs(self):
...
+class ReductionConstantDim(Reduction):
+    @classmethod
+    def chunk(cls, df, **kwargs):
+        return cls.reduction_chunk(df, **kwargs)
+
+    @classmethod
+    def combine(cls, inputs: list, **kwargs):
+        func = cls.reduction_combine or cls.reduction_aggregate or cls.reduction_chunk
+        df = _concat(inputs)
+        return func(df, **kwargs)
```
Comment (rjzamora) [inline dask_expr/reductions.py:380]: If I understand correctly, the only difference is that we don't want to convert a series back to dataframe.
Can we just add an `enforce_dataframe = True` class property to `Reduction`, and just set it to `False` for `ValueCounts`, etc?

Comment (phofl) [reply dask_expr/reductions.py:380]: Personally, I find this more readable and clear than a flag that's buried in the functions, but no strong opinion. Happy to change if you prefer

A flag should do the trick, yes

Comment (rjzamora) [reply dask_expr/reductions.py:380]: Ah, that's fair. In hindsight, I think I just find the `ReductionConstantDim` name a bit confusing. Could this be `SeriesReduction`? Or, am I misunderstanding the meaning of `ReductionConstantDim`?

Comment (phofl) [reply dask_expr/reductions.py:380]: Maybe the name is not ideal, this means that you reduce the number of rows but the dimension stays the same. Nlargest for example reduces the number of rows in your DataFrame but the result will still be a DataFrame, while sum reduces to a Series

Comment (rjzamora) [reply dask_expr/reductions.py:380]: Okay, I see. Maybe just add a description in the doc-string and I'll be fine with the name then.

Comment (phofl) [reply dask_expr/reductions.py]: Added a docstring

**Review by rjzamora — APPROVED** [review]

**Pull request merged**